components/eam/src/chemistry/mozart/lin_strat_chem.F90

```diff
@@ -313,6 +324,7 @@ subroutine linv3_strat_chem_solve(ncol, lchnk, xvmr, h2ovmr, xsfc, o3col,temp, s
     real(r8), dimension(:,:), pointer :: linoz_dPmL_dO3col
     real(r8), dimension(:,:), pointer :: linoz_cariolle_psc
     real(r8), dimension(:,:), pointer :: linoz_o3lbs
+    real(r8), dimension(:,:), pointer :: linoz_ch4_avg_srf
```

Comment:
linoz_ch4_avg_srf should be 1-dimensional for each month.

Reply:
Hi Qi @tangq . We had to follow the input workflow by the original way, 
flds(f)%input(i)%data, where data is in 3d data(:,:,:).
the variable right here refer to the data linoz_ch4_avg_srf  => fields(ch4_avg_srf_ndx)  %data(:,:,lchnk )

How about I use this reference in code?
 real(r8), pointer :: linoz_ch4_avg_srf
linoz_ch4_avg_srf  => fields(ch4_avg_srf_ndx)  %data(1,1,lchnk )

Reply:
@wlin7 , @singhbalwinder , this PR is to implement the pre-calculation of CH4 data as part of the Linoz input file to fix the related chemUCI non-BFB threading issue. Does the input data have to be 3D (time, lat, lev)? The pre-calculated CH4 input variable is 1D (time). Is there a better way to handle 1D input data? Thanks.

@jinboxie , let's get suggestions from @wlin7 or @singhbalwinder before we move forward with `fields(ch4_avg_srf_ndx) %data(1,1,lchnk )`.

Reply:
@tangq Hi Qi, For the inputdata, it is ch4_avg_srf(time). It is just when inside the code, 
the shape of readin array flds%input%data has to follow 3D. 
We had to follow this in the I/O input part (Refer to line 1293 and line 1979 in tracer.F90)